lib/chat_channel_fetcher.rb

```diff
@@ -64,6 +71,7 @@ def self.secured_public_channels(guardian, memberships, options = { following: t
           "LEFT JOIN categories ON categories.id = chat_channels.chatable_id AND chat_channels.chatable_type = 'Category'",
         )
         .where(chatable_type: ChatChannel.public_channel_chatable_types)
+        .where("chat_channels.id IN (#{generate_allowed_channel_ids_sql(guardian)})")
```

Comment:
This is done rather than having the `next if !guardian.can_see_chat_channel?(channel)` inside the decorator loops